### DISCUSSION TITLE
Is there a way to make a Global Parameterized dependencies ?

### DISCUSSION THREAD

**Discussion by rootkit7628** [fastapi/fastapi#10263] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
class CustomRouter(APIRouter):
    """Custom router."""

    def __init__(self, repository: Something = Depends(get_something)):
        """Setup router."""
        super().__init__(tags=["Customs"], prefix="/api")
        self._repository = repository

        @self.get("/{id}")
        async def get(id: int) -> ResponseBaseModel:
            """Get by id."""
            return self._repository.get(id)

        @self.post("/")
        async def create(data: CreateBaseModel) -> ResponseBaseModel:
            """Create something"""
            db_data = await self._repository.create(data)
            return ResponseBaseModel.from_orm(db_data)

        @self.patch("/{id}")
        async def update(id: int, data: UpdateBaseModel) -> ResponseBaseModel:
            """Update something"""
            db_data = await self._repository.update(id, data)
            return ResponseBaseModel.from_orm(db_data)


app = FastAPI(title="API", version="1.0.0")
app.include_router(CenterRouter(), prefix="/api")
```

### Description

I want to implement an api with this get_something dependency and it works if I define the dependency within each routes but I want to know if there is a way to define  parameterized dependencies globally for all routes without specifying this long `repository: Something = Depends(get_something)` within each route.

When I'm using the code as above, I'm getting Depends(get_something) as self._repository value because if I really understand Depends is not resolved when a request is made to an endpoint that depends on them.

It would be great if we could use like global dependencies: `APIRouter(dependencies=[{"repository": Depends(get_something)}])`

It's just about making the code more light :).

### Operating System

Linux

### FastAPI Version

0.97.0

### Pydantic Version

1.10.9

### Python Version

3.11.14

**Comment by YuriiMotov** [comment, marked as answer]
But with such approach code will be less clean and typed..

You can already solve this by creating a dependency that will use other dependencies and save their values to `Request.state`:

```python
async def (request: Request, repository: Something = Depends(get_something), other_dep: SomethingElse = Depends(get_something_else)):
    request.state._state["repository"] = repository
    request.state._state["other_dep"] = other_dep
```

Then you can access it via `Request.state` in your endpoints
